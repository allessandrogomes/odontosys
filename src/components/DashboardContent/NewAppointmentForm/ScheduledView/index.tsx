import React, { useEffect, useState } from "react"
import styles from "./styles.module.scss"
import { formatHour } from "@/utils/formatHour"
import { formatDate } from "@/utils/formatDate"

interface IScheduledView {
    durationMinutes: number | null
    dentistId: number | null
    scheduledAt: (startISO: string | null) => void
    endsAt: (endISO: string | null) => void
    onBack: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
    onNext: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
    active: boolean
}

interface ITime {
    start: string,
    end: string
}

const SELECT_DAY_AND_TIME_VIEW = "SELECT_DAY_AND_TIME_VIEW"
const SHOW_SELECTED_DAY_AND_TIME_VIEW = "SHOW_SELECTED_DAY_AND_TIME_VIEW"

export default function ScheduledView({ durationMinutes, dentistId, active, scheduledAt, endsAt, onBack, onNext }: IScheduledView) {
    const [day, setDay] = useState<string>("")
    const [times, setTimes] = useState<ITime[] | []>([])
    const [selectedTime, setSelectedTime] = useState<ITime | null>(null)
    const [currentView, setCurrentView] = useState<string>(SELECT_DAY_AND_TIME_VIEW)

    function handleNext() {
        if (selectedTime) {
            scheduledAt(selectedTime.start)
            endsAt(selectedTime.end)
            setCurrentView(SHOW_SELECTED_DAY_AND_TIME_VIEW)
        }
    }

    useEffect(() => {
        async function getAvailableTimes() {
            setSelectedTime(null)
            try {
                const response = await fetch("/api/available-times", {
                    method: "POST",
                    body: JSON.stringify({
                        date: day,
                        durationMinutes: durationMinutes,
                        dentistId: dentistId
                    })
                })

                const data = await response.json()
                setTimes(data)
            } catch (error) {
                alert(JSON.stringify(error))
            }
        }

        if (day) getAvailableTimes()
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [day])

    return (
        <div className={`${styles.box} ${active && styles.active}`}>
            {currentView === SELECT_DAY_AND_TIME_VIEW && (
                <div className={styles.scheduled}>
                    <label>Escolha o dia e hor치rio</label>
                    <input onChange={e => setDay(e.target.value)} type="date" value={day}/>
                    <div className={styles.timesBtns}>
                        {Array.isArray(times) && times.map((time, index) => <button className={`${selectedTime?.start === time.start && styles.selected}`} onClick={() => setSelectedTime(time)} key={index}>{formatHour(time.start)} - {formatHour(time.end)}</button>)}
                    </div>
                    <div className={styles.boxBtns}>
                        <button onClick={e => onBack(e)} className={styles.backBtn}>Voltar</button>
                        <button onClick={handleNext} disabled={!selectedTime} className={`${styles.nextBtn} ${selectedTime && styles.active}`}>Pr칩ximo</button>
                    </div>
                </div>
            )}

            {currentView === SHOW_SELECTED_DAY_AND_TIME_VIEW && selectedTime && day && (
                <div className={styles.showSchedule}>
                    <p>Dia e o Hor치rio selecionados: <br/><br/><span>{formatDate(day)} | {formatHour(selectedTime.start)} - {formatHour(selectedTime.end)}</span></p>
                    <div className={styles.boxBtns}>
                        <button onClick={() => setCurrentView(SELECT_DAY_AND_TIME_VIEW)} className={styles.backBtn}>Voltar</button>
                        <button onClick={e => onNext(e)} className={styles.nextBtn}>Pr칩ximo</button>
                    </div>
                </div>
            )}
        </div>
    )
}
