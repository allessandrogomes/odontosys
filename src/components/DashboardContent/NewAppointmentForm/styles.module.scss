.form {
    display: flex;
    flex-direction: column;
    width: 500px;
    height: 500px;
    box-sizing: border-box;
    padding: 30px;
    margin: 40px 0;
    border-radius: 50px;
    background: linear-gradient(145deg, #e1e1e1, #ffffff);
    box-shadow: 25px 25px 15px #d5d5d5,
        -25px -25px 15px #ffffff;

    .title {
        font-size: 2rem;
        text-align: center;
        margin: 0;
        margin-top: 20px;
    }

    .btn {
        border: none;
        background-color: var(--primary-color);
        color: var(--secondary-color);
        padding: 10px;
        border-radius: 50px;
        cursor: pointer;
        width: 100px;
        align-self: end;
        margin: auto 0 0 0;
    }

    .btnsProcedure {
        width: 100%;
        display: flex;
        justify-content: space-between;
        margin: auto 0 0 0;

        button:first-child {
            align-self: left;
        }
    }

    .confirmAppointmentView {
        display: none;
        width: 100%;
        height: 100%;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        gap: 10px;
        position: relative;

        label {
            font-size: 1.25rem;
            font-weight: 600;
        }

        p {
            margin: 0;

            span {
                font-weight: 600;
            }
        }

        .boxBtns {
            width: 100%;
            display: flex;
            justify-content: space-between;
            position: absolute;
            bottom: 0;

            button {
                border: none;
                color: var(--secondary-color);
                border-radius: 50px;
                padding: 10px;
                width: 100px;
                cursor: pointer;
            }

            .nextBtn {
                background-color: var(--primary-color);
            }

            .backBtn {
                background-color: var(--tertiary-color);
            }
        }
    }

    .confirmAppointmentView.active {
        display: flex;
    }

    .success {
        width: 100%;
        height: 100%;
        display: none;
        flex-direction: column;
        align-items: center;
        justify-content: center;

        h2 {
            text-align: center;
            font-size: 1.5rem;
            margin: 0;
            font-weight: 600;
            margin-top: 30px;
            margin-bottom: 50px;
        }

        button {
            border: none;
            background-color: var(--primary-color);
            color: var(--secondary-color);
            padding: 10px;
            border-radius: 50px;
            font-weight: 600;
            width: 80px;
            cursor: pointer;
        }
    }

    .success.active {
        display: flex;
    }
}