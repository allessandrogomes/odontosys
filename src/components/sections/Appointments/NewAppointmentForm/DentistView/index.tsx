import { useEffect, useState } from "react"
import styles from "./styles.module.scss"
import Label from "@/components/ui/Label"
import Button from "@/components/ui/Button"
import { ArrowLeft, ArrowRight } from "lucide-react"
import Spinner from "@/components/ui/Spinner"

interface IDentistView {
    procedure: string | null,
    dentistId: (id: number | null) => void
    dentistName: (dentistName: string | null) => void
    visible: boolean
    onBack: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
    onNext: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
}

const SELECT_VIEW = "SELECT_VIEW"
const SHOW_SELECTED_VIEW = "SHOW_SELECTED_VIEW"

export default function DentistView({ procedure, dentistId, dentistName, visible, onBack, onNext }: IDentistView) {
    const [dentists, setDentists] = useState<IDentist[]>([])
    const [selectedDentist, setSelectedDentist] = useState<IDentist | null>(null)
    const [currentView, setCurrentView] = useState<string>(SELECT_VIEW)
    const [isLoading, setIsLoading] = useState<boolean>(false)

    function handleNext() {
        dentistId(selectedDentist!.id)
        dentistName(selectedDentist!.name)
        setCurrentView(SHOW_SELECTED_VIEW)
    }

    function handleBack() {
        setCurrentView(SELECT_VIEW)
        dentistId(null)
        dentistName(null)
        setSelectedDentist(null)
    }

    useEffect(() => {
        async function fetchDentists() {
            setIsLoading(true)
            try {
                const response = await fetch("/api/dentist")
                const data = await response.json()
                const filteredDentists = data.filter((dentist: IDentist) => dentist.specialty.includes(procedure!))
                setDentists(filteredDentists)
            } catch (error) {
                alert(JSON.stringify(error))
            } finally {
                setIsLoading(false)
            }
        }

        if (procedure) {
            fetchDentists()
        }
    }, [procedure])
    return (
        <div className={`${styles.box} ${visible && styles.visible}`}>

            {/* Seleciona o Dentista */}
            {currentView === SELECT_VIEW && (
                <div className={styles.selectView}>
                    <Label text="Escolha o Dentista" />
                    {isLoading ?
                        <Spinner /> :
                        (
                            <div className={styles.containerDentists}>
                                {dentists && dentists.map(dentist => (
                                    <div className={`${styles.dentist} ${selectedDentist?.id === dentist.id ? styles.selected : ""}`} onClick={() => setSelectedDentist(dentist)} key={dentist.id}>
                                        <p>Dr. {dentist.name}</p>
                                    </div>
                                ))}
                            </div>
                        )}
                    <div className={styles.boxBtns}>
                        <Button onClick={e => onBack(e)} text="Voltar" iconStart={<ArrowLeft />} />
                        <Button
                            text="Próximo"
                            disabled={!selectedDentist}
                            onClick={handleNext}
                            iconEnd={<ArrowRight />}
                        />
                    </div>
                </div>
            )}

            {/* Mostra o Dentista selecionado */}
            {currentView === SHOW_SELECTED_VIEW && (
                <div className={styles.showSelectedView}>
                    <p>Dentista selecionado: <br /><span className={styles.span}>Dr. {selectedDentist!.name}</span></p>
                    <div className={styles.boxBtns}>
                        <Button text="Voltar" iconStart={<ArrowLeft />} onClick={handleBack} />
                        <Button text="Próximo" iconEnd={<ArrowRight />} onClick={e => onNext(e)} />
                    </div>
                </div>
            )}
        </div>
    )
}