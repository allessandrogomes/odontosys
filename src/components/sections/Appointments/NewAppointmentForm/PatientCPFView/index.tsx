import { useState } from "react"
import styles from "./styles.module.scss"
import { IMaskInput } from "react-imask"
import Label from "@/components/ui/Label"
import FeedbackMessage from "@/components/ui/FeedbackMessage"
import { UserX } from "lucide-react"

interface IPatientCPFView {
    onSelectPatientId: (id: number) => void
    patientName: (patientName: string | null) => void
    onNext: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
    active: boolean
}

export default function PatientCPFView({ onSelectPatientId, onNext, active, patientName }: IPatientCPFView) {
    const [cpf, setCpf] = useState<string>("")
    const [patientSelected, setPatientSelected] = useState<string | null>(null)
    const [notFound, setNotFound] = useState<boolean>(false)


    async function handleSearchPatient() {
        try {
            const response = await fetch("/api/patient", {
                method: "GET"
            })

            if (!response.ok) throw new Error("Erro ao buscar paciente")

            const data = await response.json()

            const numericCPF = cpf.replace(/\D/g, "")
            const patient = data.find((item: IPatient) => item.cpf.replace(/\D/g, "") === numericCPF)
            if (patient) {
                setPatientSelected(patient.name)
                onSelectPatientId(patient.id)
                patientName(patient.name)
                setNotFound(false)
            } else {
                setNotFound(true)
            }
        } catch (error) {
            alert(error)
        }
    }

    return (
        <div className={`${styles.box} ${active && styles.active}`}>

            {/* Campo para buscar paciente pelo CPF */}
            {!patientSelected && (
                <div className={styles.cpfField}>
                    <Label text="CPF do Paciente" />
                    <IMaskInput
                        className="imask-input"
                        mask="000.000.000-00"
                        value={cpf}
                        onAccept={(value) => setCpf(value)}
                        overwrite
                        minLength={14}
                        required
                    />
                    {notFound && <div className={styles.message}><FeedbackMessage message="Nenhum paciente encontrado" icon={<UserX />} /></div>}
                    <div className={styles.btns}>
                        <button className={`${styles.nextBtn} ${cpf.length === 14 ? styles.active : ""}`} disabled={!(cpf.length === 14)} onClick={handleSearchPatient}>Próximo</button>
                    </div>
                </div>
            )}

            {/* Mostra Paciente selecionado */}
            {patientSelected && (
                <div className={styles.selectedView}>
                    <p>Paciente selecionado: <br /><span>{patientSelected}</span></p>
                    <div className={styles.btns}>
                        <button className={styles.backBtn} onClick={() => setPatientSelected(null)}>Voltar</button>
                        <button className={styles.nextBtn} onClick={e => onNext(e)}>Próximo</button>
                    </div>
                </div>
            )}
        </div>
    )
}