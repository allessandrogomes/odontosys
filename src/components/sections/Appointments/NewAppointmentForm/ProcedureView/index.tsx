import React, { useEffect, useState } from "react"
import styles from "./styles.module.scss"
import Label from "@/components/ui/Label"
import Button from "@/components/ui/Button"
import { ArrowLeft, ArrowRight } from "lucide-react"

interface IProcedure {
    id: number | null
    procedure: string | null
    durationMinutes: number | null
}

interface IProcedureViewProps {
    onSelectProcedure: (procedure: IProcedure) => void
    onNext: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
    onBack: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
    active: boolean
}

export default function ProcedureView({ onSelectProcedure, onNext, onBack, active }: IProcedureViewProps) {
    const [procedures, setProcedures] = useState<IProcedure[]>([])
    const [selectedProcedure, setSelectedProcedure] = useState<IProcedure | null>(null)
    const [error, setError] = useState<string | null>(null)
    const [isLoading, setIsLoading] = useState<boolean>(false)

    function handleChange(e: React.ChangeEvent<HTMLSelectElement>) {
        const selectId = Number(e.target.value)
        const procedure = procedures.find(p => p.id === selectId) || null
        if (procedure) {
            onSelectProcedure(procedure)
            setSelectedProcedure(procedure)
        } else {
            onSelectProcedure({
                id: null,
                procedure: null,
                durationMinutes: null
            })
            setSelectedProcedure(null)
        }
    }

    useEffect(() => {
        async function fetchProcedures() {
            setIsLoading(true)
            
            try {
                const response = await fetch("/api/procedure")
                const data = await response.json()
                setProcedures(data)
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            } catch (error: any) {
                setError(error.message || "Erro ao carregar procedimentos. Tente novamente.")
            } finally {
                setIsLoading(false)
            }
        }

        fetchProcedures()
    }, [])

    return (
        <div className={`${styles.box} ${active && styles.active}`}>
            <Label text="Escolha o Procedimento"/>
            <select onChange={handleChange} value={selectedProcedure ? selectedProcedure.id! : ""}>
                <option disabled value="">Selecione</option>
                {procedures.map(item => <option key={item.id} value={item.id!}>{item.procedure}</option>)}
            </select>
            <div className={styles.boxBtns}>
                <Button text="Voltar" iconStart={<ArrowLeft />} onClick={e => onBack(e)} />
                <Button
                    disabled={!selectedProcedure}
                    text="PrÃ³ximo"
                    iconEnd={<ArrowRight />}
                    onClick={e => onNext(e)}
                />
            </div>
        </div>
    )
}