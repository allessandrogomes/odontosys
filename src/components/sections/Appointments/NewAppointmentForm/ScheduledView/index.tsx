import React, { useEffect, useState } from "react"
import styles from "./styles.module.scss"
import { formatHour } from "@/utils/formatHour"
import { formatDate } from "@/utils/formatDate"
import Label from "@/components/ui/Label"
import Button from "@/components/ui/Button"
import { ArrowLeft, ArrowRight, TimerOff } from "lucide-react"
import Spinner from "@/components/ui/Spinner"
import ScheduleList from "@/components/lists/ScheduleList"
import FeedbackMessage from "@/components/ui/FeedbackMessage"

interface IScheduledView {
    durationMinutes: number | null
    dentistId: number | null
    scheduledAt: (startISO: string | null) => void
    endsAt: (endISO: string | null) => void
    onBack: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
    onNext: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
    active: boolean
}

interface ITime {
    start: string,
    end: string
}

const SELECT_DAY_AND_TIME_VIEW = "SELECT_DAY_AND_TIME_VIEW"
const SHOW_SELECTED_DAY_AND_TIME_VIEW = "SHOW_SELECTED_DAY_AND_TIME_VIEW"

export default function ScheduledView({ durationMinutes, dentistId, active, scheduledAt, endsAt, onBack, onNext }: IScheduledView) {
    const [day, setDay] = useState<string>("")
    const [times, setTimes] = useState<ITime[] | []>([])
    const [isLoading, setIsLoading] = useState<boolean>(false)
    const [error, setError] = useState<string | null>(null)
    const [selectedTime, setSelectedTime] = useState<ITime | null>(null)
    const [currentView, setCurrentView] = useState<string>(SELECT_DAY_AND_TIME_VIEW)

    function handleNext() {
        if (selectedTime) {
            scheduledAt(selectedTime.start)
            endsAt(selectedTime.end)
            setCurrentView(SHOW_SELECTED_DAY_AND_TIME_VIEW)
        }
    }

    useEffect(() => {
        async function getAvailableTimes() {
            setIsLoading(true)
            setSelectedTime(null)
            try {
                const response = await fetch("/api/available-times", {
                    method: "POST",
                    body: JSON.stringify({
                        date: day,
                        durationMinutes: durationMinutes,
                        dentistId: dentistId
                    })
                })

                const data = await response.json()

                if (!response.ok) throw new Error(data.error || "Erro ao buscar os horários disponíveis")

                setTimes(data)
                // eslint-disable-next-line @typescript-eslint/no-explicit-any
            } catch (error: any) {
                setError(error.message)
            } finally {
                setIsLoading(false)
            }
        }

        if (day) getAvailableTimes()
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [day])

    return (
        <div className={`${styles.box} ${active && styles.active}`}>
            {/* Tela para selecionar o dia e horário */}
            {currentView === SELECT_DAY_AND_TIME_VIEW && (
                <div className={styles.scheduled}>
                    <Label text="Escolha o dia e horário" />
                    <input onChange={e => setDay(e.target.value)} type="date" value={day} />

                    {isLoading ? <Spinner className={styles.spinner} />
                        : error ? <FeedbackMessage message={error} />
                            : times.length > 0 ? <ScheduleList selectedSchedule={selectedTime} schedules={times} onSelectSchedule={schedule => setSelectedTime(schedule)} />
                                : <FeedbackMessage message="Nenhum horário disponível" icon={<TimerOff />}/>
                    }

                    <div className={styles.boxBtns}>
                        <Button text="Voltar" iconStart={<ArrowLeft />} onClick={e => onBack(e)} />
                        <Button text="Próximo" iconEnd={<ArrowRight />} onClick={handleNext} disabled={!selectedTime} />
                    </div>
                </div>
            )}

            {/* Tela que mostra o dia e horário escolhido */}
            {currentView === SHOW_SELECTED_DAY_AND_TIME_VIEW && selectedTime && day && (
                <div className={styles.showSchedule}>
                    <p>Dia e o Horário selecionados: <br /><br /><span>{formatDate(day)} | {formatHour(selectedTime.start)} - {formatHour(selectedTime.end)}</span></p>
                    <div className={styles.boxBtns}>
                        <Button onClick={() => setCurrentView(SELECT_DAY_AND_TIME_VIEW)} iconStart={<ArrowLeft />} text="Voltar" />
                        <Button onClick={e => onNext(e)} iconEnd={<ArrowRight />} text="Próximo" />
                    </div>
                </div>
            )}
        </div>
    )
}
