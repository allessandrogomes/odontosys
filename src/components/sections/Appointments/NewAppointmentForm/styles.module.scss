.btn {
    border: none;
    background-color: var(--primary-color);
    color: var(--secondary-color);
    padding: 10px;
    border-radius: 50px;
    cursor: pointer;
    width: 100px;
    align-self: end;
    margin: auto 0 0 0;
}

.btnsProcedure {
    width: 100%;
    display: flex;
    justify-content: space-between;
    margin: auto 0 0 0;

    button:first-child {
        align-self: left;
    }
}

.confirmAppointmentView {
    display: none;
    width: 500px;
    height: 300px;
    flex-direction: column;
    align-items: center;
    position: relative;

    p {
        margin: 0;

        span {
            font-weight: 600;
        }
    }

    .boxBtns {
        width: 100%;
        display: flex;
        justify-content: space-between;
        position: absolute;
        bottom: 0;

        .spinner {
            position: absolute;
            left: 225px;
        }

        button {
            width: 120px;
        }
    }
}

.confirmAppointmentView.active {
    display: flex;
}

.success {
    width: 100%;
    height: 100%;
    display: none;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    h2 {
        text-align: center;
        font-size: 1.5rem;
        margin: 0;
        font-weight: 600;
        margin-top: 30px;
        margin-bottom: 50px;
    }

    button {
        border: none;
        background-color: var(--primary-color);
        color: var(--secondary-color);
        padding: 10px;
        border-radius: 50px;
        font-weight: 600;
        width: 80px;
        cursor: pointer;
    }
}

.success.active {
    display: flex;
}