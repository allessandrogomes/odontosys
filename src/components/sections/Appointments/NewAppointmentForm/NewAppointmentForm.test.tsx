import { render } from '@testing-library/react'
import "@testing-library/jest-dom"
import NewAppointmentForm from '.'

jest.mock("@/components/ui/Label", () => {
    const MockLabel = () => <div>Label</div>
    MockLabel.displayName = "Label"
    return MockLabel
})
jest.mock("@/components/ui/Button", () => {
    const MockButton = ({ text, iconStart, ...props }: { text?: React.ReactNode; iconStart?: React.ReactNode } & React.ButtonHTMLAttributes<HTMLButtonElement>) => (
        <button {...props}>
            {iconStart && <span data-testid="icon-start">{iconStart}</span>}
            {text}
        </button>
    )
    MockButton.displayName = "Button"
    return MockButton
})
jest.mock("@/components/ui/Card", () => {
    const MockCard = ({ children }: { children?: React.ReactNode }) => <div>{children}</div>
    MockCard.displayName = "Card"
    return MockCard
})
jest.mock("@/components/ui/Spinner", () => {
    const MockSpinner = () => <div>Spinner</div>
    MockSpinner.displayName = "Spinner"
    return MockSpinner
})
jest.mock("@/components/layout/SectionWrapper", () => {
    const MockSectionWrapper = ({ children }: { children?: React.ReactNode }) => <section>{children}</section>
    MockSectionWrapper.displayName = "SectionWrapper"
    return MockSectionWrapper
})

// Mockando views internas para isolar a renderização inicial
jest.mock("./PatientCPFView", () => {
    const MockPatientCPFView = () => <div data-testid="PatientCPFView">PatientCPFView</div>
    MockPatientCPFView.displayName = "PatientCPFView"
    return MockPatientCPFView
})
jest.mock("./ProcedureView", () => {
    const MockProcedureView = () => <div data-testid="ProcedureView">ProcedureView</div>
    MockProcedureView.displayName = "ProcedureView"
    return MockProcedureView
})
jest.mock("./DentistView", () => {
    const MockDentistView = () => <div data-testid="DentistView">DentistView</div>
    MockDentistView.displayName = "DentistView"
    return MockDentistView
})
jest.mock("./ScheduledView", () => {    
    const MockScheduledView = () => <div data-testid="ScheduledView">ScheduledView</div>
    MockScheduledView.displayName = "ScheduledView"
    return MockScheduledView
})

describe("NewAppointmentForm", () => {
    it("deve renderizar a tela inicial corretamente (snapshot)", () => {
        const { container } = render(<NewAppointmentForm />)
        expect(container).toMatchSnapshot()
    })
})