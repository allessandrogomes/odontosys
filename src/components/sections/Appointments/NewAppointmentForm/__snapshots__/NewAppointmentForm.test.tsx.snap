// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NewAppointmentForm deve renderizar a tela inicial corretamente (snapshot) 1`] = `
<div>
  <section>
    <div
      data-testid="PatientCPFView"
    >
      PatientCPFView
    </div>
    <div
      data-testid="ProcedureView"
    >
      ProcedureView
    </div>
    <div
      data-testid="DentistView"
    >
      DentistView
    </div>
    <div
      data-testid="ScheduledView"
    >
      ScheduledView
    </div>
    <div
      class="confirmAppointmentView false"
    >
      <div>
        <div>
          Label
        </div>
        <p>
          Paciente: 
          <span />
        </p>
        <p>
          Procedimento: 
          <span />
        </p>
        <p>
          Dentista: 
          <span>
            Dr. 
          </span>
        </p>
        <p>
          Dia: 
          <span>
            31/12/1969
          </span>
        </p>
        <p>
          Horário: 
          <span>
             - 
          </span>
        </p>
        <p>
          Duração: 
          <span>
             Minutos
          </span>
        </p>
      </div>
      <div
        class="boxBtns"
      >
        <button>
          <span
            data-testid="icon-start"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-arrow-left"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m12 19-7-7 7-7"
              />
              <path
                d="M19 12H5"
              />
            </svg>
          </span>
          Voltar
        </button>
        <button>
          <span
            data-testid="icon-start"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-calendar-plus"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 19h6"
              />
              <path
                d="M16 2v4"
              />
              <path
                d="M19 16v6"
              />
              <path
                d="M21 12.598V6a2 2 0 0 0-2-2H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h8.5"
              />
              <path
                d="M3 10h18"
              />
              <path
                d="M8 2v4"
              />
            </svg>
          </span>
          Agendar
        </button>
      </div>
    </div>
    <div
      class="success false"
    >
      <h2>
        Agendamento realizado com sucesso!
      </h2>
      <button>
        OK
      </button>
    </div>
  </section>
</div>
`;
